app/assets/javascripts/discourse/app/lib/hashtag-autocomplete.js

```diff
@@ -62,7 +58,65 @@ export function hashtagTriggerRule(textarea, opts) {
   return true;
 }
 
-function _setupExperimental(context, $textArea, siteSettings, afterComplete) {
+const checkedHashtags = new Set();
+let seenHashtags = {};
```

Comment:
small concern about this ever growing in the context of an app, and more importantly in the context of tests

Reply:
Yeah understand your concern, generally I don't like adding more and more args either, though if we need to we can revisit + refactor this later, I don't see it needing more arguments for a while

Reply:
just need to make sure it's not leaking from one state to another at least

Reply:
Rereading your comment, I think you misunderstood my comment.

Im talking of `let seenHashtags = {};` this will get filled over and over and never cleared AFAICT

Reply:
Ohhh I see what you mean. That must be the case for the old category lookup code too. I will try think of something to clean this up. This would be a problem within the app if someone never refreshed and did lots of hashtag lookups, which is not really likely. But yeah definitely a concern for QUnit tests.